components/resolved_ts/tests/integrations/mod.rs

```diff
@@ -91,3 +91,40 @@ fn test_dynamic_change_advance_ts_interval() {
 
     suite.stop();
 }
+
+#[test]
+fn test_report_min_resolved_ts() {
+    fail::cfg("mock_tick_interval", "return(0)").unwrap();
+    fail::cfg("mock_collect_tick_interval", "return(0)").unwrap();
+    let mut suite = TestSuite::new(1);
+    let region = suite.cluster.get_region(&[]);
+    let ts1 = suite.cluster.pd_client.get_min_resolved_ts();
+
+    // Prewrite
+    let (k, v) = (b"k1", b"v");
+    let start_ts = block_on(suite.cluster.pd_client.get_tso()).unwrap();
+    let mut mutation = Mutation::default();
+    mutation.set_op(Op::Put);
+    mutation.key = k.to_vec();
+    mutation.value = v.to_vec();
+    suite.must_kv_prewrite(region.id, vec![mutation], k.to_vec(), start_ts);
+
+    // resolved-ts should be less than all unapplied commit ts
+    let ts2 = suite.cluster.pd_client.get_min_resolved_ts();
+    assert!(ts2 == ts1);
+    assert!(TimeStamp::new(ts2) < start_ts);
```

Comment:
Actually, `resolved_ts` only promise that it will be less than any future `commit_ts`, namely ts that get after `must_kv_prewrite`, both these two assert here can be false if the `resolved_ts` is updated after L105 and before L110.